tests/Feature/JwtAuthorizationTest.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+
+namespace Marketredesign\MrdAuth0Laravel\Tests\Feature;
+
+use Closure;
+use Illuminate\Foundation\Application;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Route;
+use Illuminate\Testing\TestResponse;
+use Marketredesign\MrdAuth0Laravel\Tests\TestCase;
+
+class JwtAuthorizationTest extends TestCase
+{
+    private const ROUTE_URI = 'test';
+
+    /**
+     * Define environment setup.
+     *
+     * @param  Application  $app
+     * @return void
+     */
+    protected function getEnvironmentSetUp($app)
+    {
+        // Set the Laravel Auth0 config values which are used to some values.
+        $app['config']->set('laravel-auth0', [
+            'domain'     => 'auth.marketredesign.com',
+            'client_id'  => '123',
+        ]);
+    }
+
+    /**
+     * Perform a GET request to the test endpoint, optionally including a Bearer token in the Authorization header.
+     *
+     * @param bool $includeBearer Whether or not to include a bearer token in the request.
+     * @param Closure|null $responseHandler Response handler for endpoint. When null, a JSON containing `test` will be
+     * returned.
+     * @return TestResponse
+     */
+    private function request(bool $includeBearer, string $scope = '', ?Closure $responseHandler = null)
+    {
+        // Define a very simply testing endpoint, protected by the jwt middleware.
+        Route::middleware('jwt' . (empty($scope) ? '' : ":$scope"))
+            ->get(self::ROUTE_URI, $responseHandler ?? function () {
+                return response()->json('test');
+            });
+
+        $headers = [];
+
+        if ($includeBearer) {
+            $headers['Authorization'] = 'Bearer token';
+        }
+
+        return $this->getJson(self::ROUTE_URI, $headers);
+    }
+
+    /**
+     * Verify the request is unauthorized when no bearer token is specified.
+     */
+    public function testBearerTokenMissing()
+    {
+        $this->mockAuth0Service(null);
+
+        $this->request(false)
+            ->assertUnauthorized()
+            ->assertSee('Bearer token missing');
+    }
+
+    /**
+     * Verify the request is unauthorized when the bearer token is invalid.
```

Comment:
And what if it is not null but still invalid (not adhering to the jwt standard)?

Reply:
Hahaha I believe you asked the exact same question when I implemented this test in the psmanager. The underlying jwt library determines whether a token is invalid. However, that library is mocked out. By setting the jwt to `null` in this case we do not actually send a `null` token, but rather immediately trigger the invalidTokenException which the library would throw as well. Thus, the only assumption we are making here is that the underlying library is correct and throws the correct exception (which is the best we can do anyway).

Reply:
Oops. In my defense I'm very busy.